lib/index.js

```diff
@@ -151,6 +156,39 @@ function lintFile(css, options, filename) {
   return processor.process(css, options.postcss);
 }
 
+/**
+ * Add the encapsulation plugins
+ * to an existing Array of plugins.
+ *
+ * N.B. the plugins must be inserted after
+ * postcss[-easy]-import otherwise encapsulation
+ * is applied only to the entry file selectors.
+ *
+ * @param {Array} plugins Existing Array of plugins or empty Array
+ * @returns {Array} New set of plugins
+ */
+function withEncapsulation(plugins) {
+  var encapsulation = require('./encapsulation');
+  function encapsulationInjectIndex(index, plugin, currentIndex) {
+    if (
+      plugin.postcssPlugin === 'postcss-easy-import' ||
+      plugin.postcssPlugin === 'postcss-import'
+    ) {
+      index = currentIndex + 1;
+    }
+    return index;
+  }
+
+  plugins.splice(
+    plugins.reduce(encapsulationInjectIndex, plugins.length),
+    0,
+    encapsulation.resetGeneric,
+    encapsulation.resetInherited
+  );
+
+  return plugins;
+}
```

Comment:
Are you suggesting to move the reduce `callback` outside of `withEncapsulation`'s scope or just use `findIndex` so that I don't have to write tests for it?